eis/src/main/java/at/ac/tgm/controller/AmpelController.java

```diff
@@ -0,0 +1,126 @@
+                .orElseThrow(() -> new RuntimeException("Lesson not found with id=" + dto.getLessonId()));
+
+        // 2) Student laden
+        Student student = studentRepository.findById(dto.getStudentId())
+                .orElseThrow(() -> new RuntimeException("Student not found with id=" + dto.getStudentId()));
+
+        // 3) Teacher laden
+        Teacher teacher = teacherRepository.findById(dto.getTeacherId())
+                .orElseThrow(() -> new RuntimeException("Teacher not found with id=" + dto.getTeacherId()));
+
+        // 4) Neues Ampel-Objekt anlegen
+        Ampel ampel = new Ampel();
+        ampel.setLesson(lesson);
+        ampel.setStudent(student);
+        ampel.setTeacher(teacher);
+        ampel.setFarbe(AmpelFarbe.valueOf(dto.getFarbe())); // "ROT", "GELB", etc.
+        ampel.setBemerkung(dto.getBemerkung());
+        ampel.setUpdatedAt(LocalDateTime.now());
+
+        // 5) Speichern & zurückgeben
+        return ampelRepository.save(ampel);
+    }
+
+    /**
+     * PUT /api/ampel/{ampelId}
+     * Aktualisiert einen vorhandenen Ampel-Eintrag.
+     */
+
+    @Secured(Roles.LEHRER)
+    @PutMapping("/{ampelId}")
+    public Ampel updateAmpel(@PathVariable Long ampelId, @RequestBody AmpelRequestDto dto) {
+        // 1) Vorhandenes Ampel-Objekt laden
+        Ampel ampel = ampelRepository.findById(ampelId)
+                .orElseThrow(() -> new RuntimeException("Ampel not found with id=" + ampelId));
+
+        // 2) Felder aktualisieren (wenn gewünscht)
+        if (dto.getFarbe() != null) {
+            ampel.setFarbe(AmpelFarbe.valueOf(dto.getFarbe()));
+        }
+        if (dto.getBemerkung() != null) {
+            ampel.setBemerkung(dto.getBemerkung());
+        }
+        // Falls du lessonId/teacherId/studentId auch änderbar machen willst:
+        if (dto.getLessonId() != null) {
+            Lesson newLesson = lessonRepository.findById(dto.getLessonId())
+                    .orElseThrow(() -> new RuntimeException("Lesson not found with id=" + dto.getLessonId()));
+            ampel.setLesson(newLesson);
+        }
+        if (dto.getStudentId() != null) {
+            Student newStudent = studentRepository.findById(dto.getStudentId())
+                    .orElseThrow(() -> new RuntimeException("Student not found with id=" + dto.getStudentId()));
+            ampel.setStudent(newStudent);
+        }
+        if (dto.getTeacherId() != null) {
+            Teacher newTeacher = teacherRepository.findById(dto.getTeacherId())
+                    .orElseThrow(() -> new RuntimeException("Teacher not found with id=" + dto.getTeacherId()));
+            ampel.setTeacher(newTeacher);
+        }
+
+        // 3) updatedAt neu setzen
+        ampel.setUpdatedAt(LocalDateTime.now());
+
+        // 4) Speichern & zurückgeben
+        return ampelRepository.save(ampel);
+    }
+    @Secured(Roles.LEHRER)
+    @DeleteMapping
+    public ResponseEntity<?> deleteAmpel(
```

Comment:
```suggestion
    public ResponseEntity<void> deleteAmpel(
```
Bitte auch testen. <?> sollte wirklich nur dann verwendet werden, wenn 2 verschiedene Datentypen oder mehr zurückkommen können und dann wenn möglich auch <? extends Supertyp> wenn möglich.

Reply:
Das ist bei einigen Methoden so, da ich entweder eine Liste zurückbekomme, oder ein Error. Und das sind zwei verschiedene Rückgabewerte.

Reply:
Die ResponseEntity ist nur für den gültigen Fall. Im Fall einer Exception wird die Methode mit der Exception verlassen (also ist hier der Rückgabetyp unwirksam) und die Expetion mit einem ExceptionHandler auf eine eigene ResponseEntity<String> gemappt. Geh bitte nochmal alle `ResponseEntity<?>` durch und schau wo dus wirklich brauchst. Hier gibst du ein noContent zurück, was einen leeren Body hat, also `ResponseEntity<void>`. Oder du machst gleich @ResponseStatus über der Methodenkopfzeile und void als Rückgabetyp.